src/java.base/share/classes/java/lang/classfile/ClassReader.java

```diff
@@ -121,12 +118,31 @@ public sealed interface ClassReader extends ConstantPool
      * {@return the constant pool entry whose index is given at the specified
      * offset within the classfile, or null if the index at the specified
      * offset is zero}
+     *
+     * @apiNote
+     * If only a particular type of entry is expected, use {@link #readEntryOrNull(
+     * int, Class) readEntryOrNull(int, Class)}.
+     *
      * @param offset the offset of the index within the classfile
      * @throws ConstantPoolException if the index is out of range of the
      *         constant pool size
      */
     PoolEntry readEntryOrNull(int offset);
 
+    /**
+     * {@return the constant pool entry of a given type whose index is given
+     * at the specified offset within the classfile, or null if the index at
+     * the specified offset is zero}
+     *
+     * @param <T> the entry type
+     * @param offset the offset of the index within the classfile
+     * @param cls the entry type
+     * @throws ConstantPoolException if the index is out of range of the
+     *         constant pool size, or zero, or the entry is not of the given type
+     * @since 24
```

Comment:
I just noticed that these are marked `@since 24`. Am I correct that this should be `@since 23`?

Reply:
Thanks for pointing out, I was under the assumption that this patch might not come into 23. I will create an issue, and you can make a PR if you feel like contributing (doc-only changes can integrate before RDPs so no rush)

Reply:
Opened #19501.